gulpfile.js

```diff
@@ -725,6 +726,40 @@ gulp.task('generic', gulp.series('buildnumber', 'default_preferences', 'locale',
   ]);
 }));
 
+// Builds the generic production viewer that should be compatible with most
+// older HTML5 browsers.
+gulp.task('generic-es5', gulp.series('buildnumber', 'default_preferences',
+                                     'locale', function() {
```

Comment:
Here, and similar below, we could extract the `function` and give it `skipBabel` parameter so we don't have to duplicate it (unless I'm overlooking something, they seem equal aside from the `SKIP_BABEL` parameter). Something like (untested):

```javascript
function createGenericViewer(skipBabel) {
   ...
   return merge([...]);
}

gulp.task('generic', gulp.series('buildnumber', 'default_preferences', 'locale', function() {
  createGenericViewer(/* skipBabel */ true);
}));

gulp.task('generic-es5', gulp.series('buildnumber', 'default_preferences', 'locale', function() {
  createGenericViewer(/* skipBabel */ false);
}));

Reply:
> [...] they seem equal aside from the `SKIP_BABEL` parameter

For what it's worth, there's also a difference in output directories...

While you're obviously correct that it *should* be possible to reduce the size of the `gulpfile.js` changes, I'd very much like to get an OK on the general idea of this PR before spending too much time on it.

In no particular order, there's a couple of things that we'd need to consider:
 - Are we OK with changing the default to producing non-translated/non-polyfilled builds by default, and instead having to *explicitly* specify when ES5-compatible builds are wanted?
   Personally I'm hoping this will be seen as a step in the right direction, since it might also allow further modernizing of the build system.
 - ES5-compatible `generic` builds are obviously still provided, but is `api-minor` enough since these changes may require that implementers manually change what files they depend on?
   Also, are we OK with only providing pre-built ES5-compatiable files for the `generic` build target, since this simplifies things considerably?
 - What about all of the `examples`?
   Personally I'd like to avoid changing them wholesale to use the ES5-compatible files, but how should we document this (inline comments, Wiki entry, and/or something else; and will people even bother to read it)?

Reply:
I'm personally also in favor of preferring ES6 builds over ES5 builds. However, I'm not sure if `[api-minor]` is the right classification for this change since it's quite a behavior change; it feels more like `[api-major]` to me.

I'm thinking that to make this work for `[api-minor]` we can do the reverse, namely keep the default at ES5 but also provide ES6 builds. This is already a step in the right direction because now we don't really provide ES6 builds, so it will allow people to start using it and provide us with feedback. Moreover, we and users don't need to adjust anything immediately. For the next major version, we can then swap the builds, making ES6 the default and ES5 the alternative option. In the major version after that, we can then hopefully drop the ES5-only build altogether.

What do you think? Also /cc @yurydelendik @brendandahl for opinions on this.

Reply:
> I'm thinking that to make this work for `[api-minor]` we can do the reverse, namely keep the default at ES5 but also provide ES6 builds.

Honestly, this feels like *way* more work/complication compared to the current approach (and certainly not something I'd bother doing, unless the users complaining about there not being any non-translated builds actually compensating me for my time :-).

Even trying to do something like this would also make e.g. the `pdfjs-dist` repository really big and complicated, not to mention *even slower* to generate, since you'd probably need to essentially duplicate all of it's code which is precisely what I actively tried to avoid here.
In that case I'd personally much rather just immediately increment the `major` version, and keep the current approach of modern builds with a few exceptions, and then be done with it since this simplifies things considerably!

Furthermore, note that the non-translated builds aren't actually [ECMAScript 2015](https://en.wikipedia.org/wiki/ECMAScript#6th_Edition_-_ECMAScript_2015) (i.e. older classification ES6) but rather at least [ECMAScript 2018](https://en.wikipedia.org/wiki/ECMAScript#9th_Edition_-_ECMAScript_2018) (i.e. older classification ES9). Hence labelling something as ES6 isn't even remotely correct

Reply:
*Sorry if the above seemed a bit grumpy!*

Some additional thoughts:

> [...] so it will allow people to start using it and provide us with feedback.

Assuming that a modern, and up-to-date, standards-compliant browser is being used I really cannot imagine there being any problems with using a non-translated/non-polyfilled build (based on the fact that the PDF Viewer in Firefox has been shipping this way for years).

Also, simply incrementing the `major` version should definitely allow to move forward and "fix" things like e.g. issue #11225 
Finally, considering that the *first* official release from the `2.0` branch was released a year ago, it doesn't seem so bad to me to increase that now (especially for these kind of build improvements).

Reply:
Incrementing the major version number for the reasons you mentioned sounds like a good idea, so perhaps that's indeed an easier approach. I'll think about this plan a bit more.